### PR TITLE
Updated Find command 

### PR DESCRIPTION
**Pull request by kashing555**
Updated to align with add and edit usage.

### CODE REVIEW

File: src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -1,85 +1,96 @@
 package seedu.trackbeau.logic.parser;
 
 import static seedu.trackbeau.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.trackbeau.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
+import static seedu.trackbeau.logic.parser.CliSyntax.*;
```
Comment (flairekq) [inline src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java]: I think can change it to explicitly state all the classes you are importing to adhere to the coding guidelines :)

File: src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -1,54 +1,56 @@
...
+                        searchString = searchString + tag.tagName + " ";
+                    }
                 }
-            }
-
-            for (String keyword : keywords) {
-                if (StringUtil.containsWordIgnoreCase(searchString, keyword)) {
-                    anyMatch = true;
-                    break;
+                if (keywords != null) {
```
Comment (flairekq) [inline src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java:47]: I feel lines 47-53 has arrow-head style code, maybe you can try doing 
`if (keywords == null) { 
      continue;
  }`
  `for (String keyword : keywords) {
      ...
  }
`

File: src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -59,8 +61,21 @@ public boolean test(Customer customer) {
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof SearchContainsKeywordsPredicate // instanceof handles nulls
-                && keywords.equals(((SearchContainsKeywordsPredicate) other).keywords)); // state check
+        if (other == this) {
```
Comment (flairekq) [inline src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java:64]: Similarly, for lines 64-79 seems to have arrow-head style code, maybe you can try doing
`if (other == this) { return true; }`
`if !(other instanceof SearchContainsKeywordsPredicate) { return false; }`
`for (int i = 0; i < 9; i++) { ... }`

**Review by flairekq — CHANGES_REQUESTED** [review]
Overall looks good! Just that some areas can be changed to adhere to code quality standards.

File: src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (kashing555) [reply src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java]: Fix in previous commit

File: src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java (hunk 2 above)
Comment (kashing555) [reply src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java:47]: okay will do

File: src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java (hunk 3 above)
Comment (kashing555) [reply src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java:64]: okay will do

File: src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java [hunk 4]
```diff
@@ -19,67 +25,80 @@
...
+                            .parseName(argMultimap.getValue(PREFIX_SERVICES).get()).toString().split(" ")));
+        }
+
+        if (argMultimap.getValue(PREFIX_ALLERGIES).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(8,
+                Arrays.asList(ParserUtil
+                    .parseName(argMultimap.getValue(PREFIX_ALLERGIES).get()).toString().split(" ")));
+        }
+
+        System.out.println(prefixArr.toString());
```
Comment (hanhuiice) [inline src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java:99]: Is this print statement extra?

**Review by hanhuiice — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java [hunk 5]
```diff
@@ -19,67 +25,80 @@
...
+                            .parseName(argMultimap.getValue(PREFIX_STAFFS).get()).toString().split(" ")));
+        }
+
+        if (argMultimap.getValue(PREFIX_SERVICES).isPresent() && argMultimap.getPreamble().isEmpty()) {
+            prefixArr.add(7,
+                    Arrays.asList(ParserUtil
+                            .parseName(argMultimap.getValue(PREFIX_SERVICES).get()).toString().split(" ")));
+        }
+
+        if (argMultimap.getValue(PREFIX_ALLERGIES).isPresent() && argMultimap.getPreamble().isEmpty()) {
```
Comment (isabelteo) [inline src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java:93]: codes for these chunks are very similar - you can consider abstracting out to another method

File: src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java [hunk 6]
```diff
@@ -12,43 +13,47 @@
...
-                searchString = customer.getClass().getDeclaredMethod(searchArea).invoke(customer).toString();
-            } else {
-                // Will always return type Set<Tag> from the 3 possible methods in the Customer class.
-                @SuppressWarnings("unchecked")
-                Set<Tag> tagList = (Set<Tag>) customer.getClass().getDeclaredMethod(searchArea).invoke(customer);
-                for (Tag tag : tagList) {
-                    searchString = searchString + tag.tagName + " ";
+            for (int i = 0; i < 9; i++) {
+                List<String> keywords = keywordsList.get(i);
+                if (i < 6) {
```
Comment (isabelteo) [inline src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java:37]: magic number? for 9 in line 35 and 6 in line 37

File: src/test/java/seedu/trackbeau/logic/commands/CommandTestUtil.java [hunk 7]
```diff
@@ -172,8 +173,9 @@ public static void showCustomerAtIndex(Model model, Index targetIndex) {
 
         Customer customer = model.getFilteredCustomerList().get(targetIndex.getZeroBased());
         final String[] splitName = customer.getName().fullName.split("\\s+");
-        model.updateFilteredCustomerList(new SearchContainsKeywordsPredicate("getName",
-                0, Arrays.asList(splitName[0])));
+        ArrayList<List<String>> prefixArr = new ArrayList<List<String>>(Collections.nCopies(8, null));
```
Comment (isabelteo) [inline src/test/java/seedu/trackbeau/logic/commands/CommandTestUtil.java:176]: magic number 8?

Comment (kashing555) [reply src/test/java/seedu/trackbeau/logic/commands/CommandTestUtil.java:176]: should be 9

**Comment by hanhuiice** [comment]
LGTM

**Pull request merged**